odl/set/sets.py

```diff
@@ -217,6 +226,13 @@ def __contains__(self, other):
         characters, `False` otherwise."""
         return isinstance(other, basestring) and len(other) == self.length
 
+    def contains_all(self, array):
+        """Test if `array` is an array of strings with correct length."""
+        array = np.asarray(array)  # allow seqence type input
+        dtype_byte = np.dtype('S{}'.format(self.length))
+        dtype_uni = np.dtype('<U{}'.format(self.length))
+        return array.dtype in (dtype_byte, dtype_uni)
```

Comment:
This was obviously an attempt to duck-type and allow array-like input. Not sure if it's possible to get away without converting to an array. Perhaps digging into NumPy and looking for their dtype determination helper could be a solution. Or being strict and allowing only NumPy arrays (and meshgrids, which obviously would fail here, ahem).